packages/dds/tree/src/simple-tree/schemaFactory.ts

```diff
@@ -409,19 +430,18 @@ export class SchemaFactory<TScope extends string = string, TName extends number
 				input: ReadonlyMap<string, InsertableTreeNodeFromImplicitAllowedTypes<T>>,
 			) {
 				super(input);
+
+				const proxyTarget = customizable ? this : undefined;
+
 				if (isFlexTreeNode(input)) {
-					return createNodeProxy(
-						input,
-						customizable,
-						customizable ? this : undefined,
-					) as schema;
+					return createNodeProxy(input, customizable, proxyTarget) as schema;
 				} else {
 					const flexSchema = getFlexSchema(this.constructor as TreeNodeSchema);
 					return createRawNodeProxy(
 						flexSchema as FlexMapNodeSchema,
 						input,
 						customizable,
-						customizable ? this : undefined,
+						proxyTarget,
 					) as unknown as schema;
```

Comment:
Structurally named arrays would already preserve the [] prototype chain.

I'd propose that the prototype chain should be determined by whether `_.array` is extended, not whether it's named, but trying not to change too much at once.